### ISSUE TITLE
微调llama3.1 bug

### ISSUE THREAD

**Issue by lijinda22** [LLaVA-VL/LLaVA-NeXT#457]
请问llavanext代码库还有在维护吗, 我在使用llama3.1作为LLM模型的时候遇到非常多bug, 比如pad_token, unk_token, 模型generate报错inf, nan等, lora模型合并后推理dtype不一致等等, 请问是否有可以微调llama3.1的llava-llama3.1的开源代码库

**Comment by lijinda22** [comment]
in LlavaLlamaForCausalLM forward

```python
res = super().forward(
                input_ids=input_ids,
                attention_mask=attention_mask,
                position_ids=position_ids,
                past_key_values=past_key_values,
                inputs_embeds=inputs_embeds,
                labels=labels,
                use_cache=use_cache,
                output_attentions=output_attentions,
                output_hidden_states=output_hidden_states,
                return_dict=return_dict,
            )
print("in forward res:", res)
```
in forward res: CausalLMOutputWithPast(loss=None, logits=tensor([[[-3.0938,  1.5547,  7.3750,  ...,  1.7344,  1.7344,  1.7344],
         [-4.5938, -4.6250, -7.7500,  ...,  4.6562,  4.6562,  4.6562],
         [-6.8125, -1.8203, -1.4531,  ...,  4.4062,  4.4062,  4.4062],
         ...,
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan],
         [    nan,     nan,     nan,  ...,     nan,     nan,     nan]]],
       device='cuda:0'), past_key_values=DynamicCache(), hidden_states=None, attentions=None)

**Comment by lijinda22** [comment]
定位到问题了, 不是llama编码中nan的原因, 是clip的vision模型推理得到nan, 但是训练的时候提取的特征都是正常的, 同样是dtype为bfloat16, 目前问题还是没有解决.

**Comment by lijinda22** [comment]
found the bug!!! 使用eval/model_vqa.py推理的时候 需要load llavallama model, 在 llava/model/builder.py中, 如果model_base是none, 使model_path, 是预训练和微调后的llava模型本地路径, model = LlavaLlamaForCausalLM.from_pretrained(model_path, low_cpu_mem_usage=True, attn_implementation=attn_implementation, config=llava_cfg, **kwargs) 这里通过模型加权重加载llm模块的时候, 会对clip vision模块的参数进行错误地覆盖或修改, 希望这个bug可以有修复!!!! 训练的时候是只需要load llm的权重, 所以对vision没有影响, 然后再收到初始化视觉编码器和load权重
我打算通过类似训练的逻辑来解决这个bug, 不过这个bug没别人遇到吗....

**Comment by lijinda22** [comment]
my fault.... 粗心了, delay load即可
